util/diff_runs.py

```diff
@@ -0,0 +1,244 @@
+
+
+def main():
+    args = parse_flags()  # type: argparse.Namespace
+
+    loggingLevel = getattr(logging, args.log.upper(), None)
+    logging.basicConfig(level=loggingLevel)
+    logger = logging.getLogger()
+
+    if sys.version_info < (3,):
+        logger.fatal('python3 required')
+        return
+
+    differ = RunDiffer(args, logger, Fetcher())
+    differ.diff()
+
+
+# Create an ArgumentParser for the flags we'll expect.
+def parse_flags():  # type: () -> argparse.Namespace
+    # Re-use the docs above as the --help output.
+    parser = argparse.ArgumentParser(description=inspect.cleandoc(__doc__))
+    desc = (
+        '{platform(s)}@{sha} where platform(s) is a comma separated list of '
+        'platforms, e.g. \'chrome,safari\', and sha is SHA[0:10] of the run '
+        'to compare against the initial run'
+    )
+    parser.add_argument(
+        '--before',
+        required=True,
+        type=PlatformsAtRevision.parse,
+        help=desc)
+    parser.add_argument(
+        '--after',
+        required=True,
+        type=PlatformsAtRevision.parse,
+        help=desc)
+    parser.add_argument(
+        '--log',
+        type=str,
+        default='INFO',
+        help='Log level to output')
+    namespace = parser.parse_args()
+
+    # Check the before and after platform lists have the same length.
+    if len(namespace.before.platforms) != len(namespace.after.platforms):
+        msg = ('Different number of platforms in before and after flags.\n'
+               '--before %s\n--after %s')
+
+        raise ValueError(msg
+                         % (namespace.before.platforms,
+                            namespace.after.platforms))
+
+    return namespace
+
+
+class PlatformsAtRevision(object):
+    def __init__(self, sha, platforms):  # type: (str, List[str]) -> None
+        self.sha = sha
+        self.platforms = platforms
+
+    @classmethod
+    def parse(cls, value):  # type: (str) -> PlatformsAtRevision
+        pieces = value.split('@')
+        sha = 'latest'
+        platforms = ['chrome', 'edge', 'firefox', 'safari']
+        if len(pieces) > 2:
+            raise ValueError(value)
+
+        if len(pieces) > 1:
+            sha = pieces[1]
+        elif len(pieces) > 0:
+            sha = pieces[0]
+
+        if len(pieces) > 1:
+            platforms = pieces[0].split(',')
+
+        return PlatformsAtRevision(sha, platforms)
+
+
+class RunDiffer(object):
+    def __init__(self,
+                 args,  # type: argparse.Namespace
+                 logger,  # type: logging.Logger
+                 fetcher  # type: Fetcher
+                 ):
+        self.args = args
+        self.logger = logger
+        self.fetcher = fetcher
+
+    def diff(self):
+        beforeSHA = self.args.before.sha  # type: str
+        afterSHA = self.args.after.sha  # type: str
+
+        for i in range(0, len(self.args.after.platforms)):
+            specBefore = '%s@%s' % (self.args.before.platforms[i],
```

Comment:
We should document somewhere that `--before=foo,bar@SHA --after=baz,quz@SHA` will match `foo:baz` and `bar:quz` for diffing.

Reply:
Done.